src/common/pagination/pagination.data.ts

```diff
@@ -0,0 +1,88 @@
+const QUERY_PARAM_CURSOR = 'cursor';
+const QUERY_PARAM_LIMIT = 'limit';
+const QUERY_PARAM_OFFSET = 'offset';
+
+export class PaginationData {
+  private constructor(readonly limit?: number, readonly offset?: number) {}
+
+  /**
+   * Extracts {@link PaginationData} from {@link url}. The url
+   * must have a cursor query parameter to be extracted.
+   *
+   * This function should never fail. If the format is not the one
+   * expected it assumes that the respective pagination data does
+   * not exist
+   *
+   * @param url - url which contains the cursor query param
+   */
+  static fromCursor(url: Readonly<URL>): PaginationData {
+    // If we don't have a cursor, assume no pagination data
+    const cursorValue = url.searchParams.get(QUERY_PARAM_CURSOR) ?? '';
+    const cursorSearchParams = new URLSearchParams(cursorValue);
+    const limit = Number(cursorSearchParams.get(QUERY_PARAM_LIMIT) ?? NaN);
+    const offset = Number(cursorSearchParams.get(QUERY_PARAM_OFFSET) ?? NaN);
```

Comment:
I'm defaulting to `NaN` here as a `Number` of a an empty string is `0`. This results in incorrect `limit` and `offset` values (where `0` should be different than no value)

Reply:
Perfect! JS is pretty weird sometimes 🙂